src/common/crypto/SignatureProvider.ts

```diff
@@ -0,0 +1,82 @@
+import { SignatureVerificationException } from '../exceptions';
+import { CryptoProvider } from './CryptoProvider';
+import { NodeCryptoProvider } from './NodeCryptoProvider';
+import { SubtleCryptoProvider } from './SubtleCryptoProvider';
+
+export class SignatureProvider {
+  private cryptoProvider: CryptoProvider;
+
+  constructor(subtleCrypto?: typeof crypto.subtle) {
+    if (typeof crypto !== 'undefined' && typeof crypto.subtle !== 'undefined') {
```

Comment:
This unfortunately will cause problems in certain JS runtimes. Even though you're doing the check it'll still throw an error at runtime because `crypto` doesn't exist. Instead we need to use the [Node](https://github.com/workos/workos-node/blob/main/src/index.ts) and [Worker](https://github.com/workos/workos-node/blob/main/src/index.worker.ts) entry points to specify which version to use.